### PR TITLE
Write file to the specified outputPath

### PR DESCRIPTION
**Pull request by chkpnt**
Thanks for this little helper :-)

I've just noticed that the outputPath is currently ignored.

### CODE REVIEW

File: Sources/Core/Commons/Tools/Export+Write.swift [hunk 1]
```diff
@@ -6,7 +6,7 @@
 //
 
 public extension Export {
-    func write() -> Result<Export, Xccov.Error> {
-        self.content.write(filename: self.filename).map { self }
+    func write(toDirectory directory: String) -> Result<Export, Xccov.Error> {
```
Comment (twittemb) [inline Sources/Core/Commons/Tools/Export+Write.swift]: in order to be consistent with the existing naming (and also the Apple naming), could you rename `toDirectory directory` in `atPath path` please ?

File: Sources/Core/Commons/Tools/String+Write.swift [hunk 2]
```diff
@@ -5,9 +5,13 @@
 //  Created by Thibault Wittemberg on 2020-06-03.
 //
 
+import Foundation
+
 public extension String {
-    func write(filename: String) -> Result<Void, Xccov.Error> {
-        guard (try? self.write(toFile: filename, atomically: true, encoding: .utf8)) != nil else {
+    func write(toFile filename: String, inDirectory directory: String) -> Result<Void, Xccov.Error> {
```
Comment (twittemb) [inline Sources/Core/Commons/Tools/String+Write.swift]: same here `inDirectory directory` -> `atPath path`

**Review by twittemb — CHANGES_REQUESTED** [review]
Hi @chkpnt 

Thanks a lot for your PR and its quality 👍

Could you just address the 2 minor comments please ?

thanks again.

BTW: good catch for the outputPath ... it proves there was something missing in my coverage.

File: Sources/Core/Commons/Tools/Export+Write.swift (hunk 1 above)
Comment (chkpnt) [reply Sources/Core/Commons/Tools/Export+Write.swift]: done

File: Sources/Core/Commons/Tools/String+Write.swift (hunk 2 above)
Comment (chkpnt) [reply Sources/Core/Commons/Tools/String+Write.swift]: done

**Comment by chkpnt** [comment]
I'm glad you like the proposed change. :-)

**Comment by twittemb** [comment]
@chkpnt 

On my Mac I have a compile error in the tests for the extension `uniqueTemporaryDirectory` because it uses `temporaryDirectory` that is available only since macOS 10.12.

Can you add this in the Package.swift (after the name) in order to set the minimum os version compatible with xcc. I guess macOS 10.12 is old enough so we can use it as the minimum compatible version.

```
    platforms: [
        .macOS(.v10_12),
    ],
```

**Comment by chkpnt** [comment]
I've added the missing platfom field.

But I'm curious: How did you figure that out? I mean.... if your MacOS is that old that it didn't compile, then it won't compile even if there is the platform described in `Package.swift`, will it?

(I've also rewritten the commit message to be more consistent with your history.)

**Review by twittemb — APPROVED** [review]

**Comment by twittemb** [comment]
@chkpnt 

my Mac runs Catalina ... so I don't know why Xcode complained about that 🤔

Thanks again for your PR and for having adapted the commit message to my standard, not all contributors pay attention to that.

Merged.

**Comment by chkpnt** [comment]
> my Mac runs Catalina ... so I don't know why Xcode complained about that 🤔

Oh.... lol... I just thought opening / importing the project with Xcode is too much hassle as there is no `*.xcodeproj` in the repo. As I never created a Swift PM so far (I only used them up to now), I didn't know there is simply no `*.xcodeproj` for Swift PMs. 🤦  (I simply opened the folder in Visual Studio Code and used the `Makefile` to compile and run the tests.)

**Pull request merged**